### PR TITLE
Drop 1.9.3 support

### PR DESCRIPTION
**Pull request by tamird**
@sodabrew I might've missed something, but doesn't look like there's a huge upshot to doing this (at least not until we're ready to drop 2.0.0).

### CODE REVIEW

File: ext/mysql2/client.h [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #ifndef MYSQL2_CLIENT_H
 #define MYSQL2_CLIENT_H
 
+// rb_thread_call_without_gvl was added in 2.0.0. See:
+// https://github.com/ruby/ruby/commit/c51a826.
```
Comment (sodabrew) [inline ext/mysql2/client.h]: Could be removed? Or is it needed for RBX?

**Comment by sodabrew** [comment]
Can remove `if Thread.respond_to?(:handle_interrupt)` blocks in client.rb and statement.rb, too.

**Comment by tamird** [comment]
Duh, good eye. Removed some more things.

**Comment by sodabrew** [comment]
This matches up with all the changes I was testing locally, thank you!

**Comment by tamird** [comment]
Yikes :(

**Comment by sodabrew** [comment]
_I noticed a few more `if... handle interrupts?` in the specs._

**Comment by tamird** [comment]
Just saw one, done.

**Comment by tamird** [comment]
Interesting that RHSCL also dropped support for 2.0 at the same time they dropped 1.9. Shall we keep this train rolling?

**Comment by sodabrew** [comment]
Just a quick inventory suggests that only Ubuntu Trusty and Debian Wheezy are holding out at 1.9.3/2.0.0. Wheezy EOL is May 2018, and Trusty EOL is April 2019 (ugh, as mentioned, but also by that time will be two full LTS releases behind, so I'm less worried).

Older Mac OS X is at Ruby 2.0, Debian Jessie is on Ruby 2.1, and meanwhile everyone else has congregated at Ruby 2.3 now: Debian Squeeze and Buster, Ubuntu Xenial and (probably) Bionic, OS X High Sierra, and RHSCL 7.

**Pull request merged**